extensions/podman/packages/extension/src/certificate-detection/certificate-detection-service.ts

```diff
@@ -0,0 +1,243 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Dirent } from 'node:fs';
+import * as fs from 'node:fs/promises';
+import * as os from 'node:os';
+import * as path from 'node:path';
+
+import * as extensionApi from '@podman-desktop/api';
+
+export interface CertificateDetectionResult {
+  hasCustomCertificates: boolean;
+  certificateCount: number;
+  scanDurationMs: number;
+  isWindows?: boolean;
+  isMac?: boolean;
+  isLinux?: boolean;
+  errors?: DetectionError[];
+}
+
+export interface DetectionError {
+  error: string;
+  code?: string;
+}
+
+export interface CertificateDetectionConfig {
+  enableTelemetry: boolean;
+  scanTimeoutMs: number;
+  excludedExtensions: string[];
+  validCertExtensions: string[];
+  certDirectories: { system: string[]; user: string[] };
+  followSymlinks: boolean;
+}
+
+export const DEFAULT_CONFIG: CertificateDetectionConfig = {
+  enableTelemetry: true,
+  scanTimeoutMs: 10000,
+  excludedExtensions: ['.pem', '.cer'],
+  validCertExtensions: ['.crt', '.cert', '.key'],
+  certDirectories: {
+    system: ['/etc/containers/certs.d'],
+    user: ['$HOME/.config/containers/certs.d'],
+  },
+  followSymlinks: false,
+};
+
+export class CertificateDetectionService {
+  private config: CertificateDetectionConfig;
+  private readonly telemetryLogger?: extensionApi.TelemetryLogger;
+
+  constructor(telemetryLogger?: extensionApi.TelemetryLogger, config?: Partial<CertificateDetectionConfig>) {
+    this.telemetryLogger = telemetryLogger;
+    this.config = { ...DEFAULT_CONFIG, ...config };
+  }
+
+  async detectCustomCertificates(): Promise<CertificateDetectionResult> {
+    const startTime = Date.now();
+
+    try {
+      const result = await this.performDetection();
+      result.scanDurationMs = Date.now() - startTime;
+
+      if (this.config.enableTelemetry && this.telemetryLogger) {
+        await this.sendTelemetry(result);
+      }
```

Comment:
I would move the compute in the if telemetry block

because why collect data if we don't need it ?

it'll simplify the logic as well in the catch as we check again if telemetry is enabled